Fixing a few typos and style issues (#11883)
apache/druid: docs/querying/dimensionspecs.md

## DimensionSpec

> A `DimensionSpec` defines how dimension values get transformed prior to aggregation.

Reviewer: ```suggestion
A `DimensionSpec` defines how to transform dimension values prior to aggregation.
```

}
```

> When specifying a `DimensionSpec` on a numeric column, you should include the type of the column in the `outputType` field. If left unspecified, the `outputType` defaults to STRING.

Reviewer: ```suggestion
When specifying a `DimensionSpec` on a numeric column, you should include the type of the column in the `outputType` field. The `outputType` defaults to STRING when not specified.
```

}
```

> `outputType` may also be specified in an `ExtractionDimensionSpec` to apply type conversion to results before merging. If left unspecified, the `outputType` defaults to STRING.

Reviewer: ```suggestion
You can specify an `outputType` in an `ExtractionDimensionSpec` to apply type conversion to results before merging. The `outputType` defaults to STRING when not specified.
```

Please refer to the [Output Types](#output-types) section for more details.

### Filtered DimensionSpecs

A filtered `DimensionSpec` is only useful for multi-value dimensions. Say you have a row in Apache Druid that has a multi-value dimension with values ["v1", "v2", "v3"] and you send a groupBy/topN query grouping by that dimension with a [query filter](filters.md) for a value of "v1". In the response you will get 3 rows containing "v1", "v2" and "v3". This behavior might be unintuitive for some use cases.

> This happens because "query filter" is internally used on the bitmaps and only used to match the row to be included in the query result processing. With multi-value dimensions, "query filter" behaves like a contains check, which will match the row with dimension value ["v1", "v2", "v3"]. 

Reviewer: ```suggestion
This happens because Druid uses the "query filter" internally used on the bitmaps only to match the row to include in query result processing. With multi-value dimensions, "query filter" behaves like a contains check that matches the row with dimension value ["v1", "v2", "v3"]. 
```
make sure I didn't mangle the meaning here. This whole bit on filtered dimensions is a little confusing.

Author: I think the final clause is non-defining, so it should still be ", which matches ..."  The first sentence is confusing but I think it could accurately say, "This happens because Druid uses the "query filter" internally on bitmaps only to match the row...." Will edit separately

> > See the section on "Multi-value columns" in [segment](../design/segments.md) for more details.

Reviewer: ```suggestion
See the section on "Multi-value columns" in [segment](../design/segments.md) for more details.
```

Then the groupBy/topN processing pipeline "explodes" all multi-value dimensions resulting 3 rows for "v1", "v2" and "v3" each.

In addition to "query filter", which efficiently selects the rows to be processed, you can use the filtered dimension spec to filter for specific values within the values of a multi-value dimension. These dimension specs take a delegate `DimensionSpec` and a filtering criteria. From the "exploded" rows, only rows matching the given filtering criteria are returned in the query result.

> The following filtered dimension spec acts as a whitelist or blacklist for values as per the `isWhitelist` attribute value.

Reviewer: ```suggestion
The following filtered dimension spec defines the values to include or exclude as per the `isWhitelist` attribute value.
```
trying to avoid whitelist/blacklist even though it's unavoidable in the code.

```json
{ "type" : "listFiltered", "delegate" : <dimensionSpec>, "values": <array of strings>, "isWhitelist": <optional attribute for true/false, default is true> }
```

> The following filtered dimension spec retains only the values matching regex. Note that `listFiltered` is faster than this and one should use that for whitelist or blacklist use case.

Reviewer: ```suggestion
The following filtered dimension spec retains only the values matching a regex.  You should use the `listFiltered` function for inclusion and exclusion use cases because it is faster.
```
